pkg/controller/controller.go

```diff
@@ -123,11 +123,23 @@ func (ctrl *resizeController) updatePVC(oldObj, newObj interface{}) {
 	newSize := newPVC.Spec.Resources.Requests[v1.ResourceStorage]
 	oldSize := oldPVC.Spec.Resources.Requests[v1.ResourceStorage]
 
+	newResizerName := newPVC.Annotations[util.VolumeResizerKey]
+	oldResizerName := oldPVC.Annotations[util.VolumeResizerKey]
+
 	// We perform additional checks to avoid double processing of PVCs, as we will also receive Update event when:
 	// 1. Administrator or users may introduce other changes(such as add labels, modify annotations, etc.)
 	//    unrelated to volume resize.
 	// 2. Informer will resync and send Update event periodically without any changes.
-	if newSize.Cmp(oldSize) > 0 {
+	//
+	// We add the PVC into work queue when the new size is larger then the old size
+	// or when the resizer name annotation is missing from old PVC but present in the new PVC.
+	// This is needed for CSI migration since the first time a migrated PVC is expanded,
+	// it does not yet have the annotation because annotation is only added by in-tree resizer when it receives a volume
+	// expansion request. So first update event that will be received by external-resizer will be ignored because it won't
+	// know how to support resizing of a "un-annotated" in-tree PVC. When in-tree resizer does add the annotation, a second
+	// update even will be received and we add the pvc to workqueue. If annotation matches the registered driver name in
+	// csi_resizer object, we proceeds with expansion internally or we discard the PVC.
+	if newSize.Cmp(oldSize) > 0 || (newResizerName != "" && oldResizerName == "") {
```

Comment:
Given https://github.com/kubernetes-csi/external-resizer/pull/39#discussion_r296868667 I think it makes sense to check if `newResizerName != oldResizeName`

Reply:
Updated